### PR TITLE
Run spotless before the release version updates are committed

### LINKED ISSUE #10602: Maven release plugin doesn't work with sort plugin
**Describe the bug**

The release plugin updates the version in the pom file during the jenkins release pipeline. It will change some formating of the pom and will cause the sort plugin to fail. See related PR. https://github.com/camunda/zeebe/actions/runs/3180846815/jobs/5184904296

```
Error:  Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.27.1:check (spotless-check) on project zeebe-bom: The following files had format violations:
Error:      pom.xml
Error:          @@ -8,7 +8,7 @@
Error:           ····<artifactId>camunda-release-parent</artifactId>
Error:           ····<version>3.9.1</version>
Error:           ····<!--·do·not·remove·empty·tag·-·[http://jira.codehaus.org/browse/MNG-4687·--](http://jira.codehaus.org/browse/MNG-4687%C2%B7--)>
Error:          -····<relativePath·/>
Error:          +····<relativePath></relativePath>
Error:           ··</parent>
Error:           
Error:           ··<groupId>io.camunda</groupId>
Error:          @@ -128,7 +128,7 @@
Error:           ··········<configuration>
Error:           ············<pom>
Error:           ··············<!--·[https://github.com/diffplug/spotless/blob/main/plugin-maven/README.md#sortpom·--](https://github.com/diffplug/spotless/blob/main/plugin-maven/README.md#sortpom%C2%B7--)>
Error:          -··············<sortPom·/>
Error:          +··············<sortPom></sortPom>
Error:           ············</pom>
Error:           ··········</configuration>
Error:           ········</plugin>
Error:  Run 'mvn spotless:apply' to fix these violations.
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error: Process completed with exit code 1.
```

See related change which causes this https://github.com/camunda/zeebe/pull/10599/commits/e8e98560ab7123d3cdbdd4fdf63b5934cd251813#diff-954221b717ff3d93b2f171be5b7d728359dfa4ebee98e8e5691c0697682d247eL131

**To Reproduce**

Build a release

**Expected behavior**
No issues with sort plugin

**Log/Stacktrace**
https://github.com/camunda/zeebe/actions/runs/3180846815/jobs/5184904296

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

The release plugin updates the versions in our pom files. This update will break the formatting that we have configured in spotless. As a result, when we'd like to merge the release branch back to the main branch, we manually need to run spotless:apply in order to restore the formatting.

We can add a completion goal to the release plugin. This parameter will:

["Goals to run on completion of the preparation step, after transformation back to the next development version but before committing. Space delimited."](https://maven.apache.org/maven-release/maven-release-plugin/prepare-mojo.html#completiongoals)

This means it will execute `spotless:apply` once the release plugin has updated the versions to the next development version, but before the commit is made. Making the correct format part of the release commit.

The comments [here](https://github.com/camunda/zeebe/issues/10602#issuecomment-1329376912) mention the deletion of the `-DskipChecks=true`

  
  **Manual test**

Please see the example branch here: https://github.com/camunda/zeebe/commits/10602_maven_release_spotless_example

The [[maven-release-plugin] prepare release 0.0.0](https://github.com/camunda/zeebe/commit/8bec9c74d18daffed811db320067a35e0e8e4314) commit messes up the formatting of the pom file
In the [[maven-release-plugin] prepare for next development iteration](https://github.com/camunda/zeebe/commit/b08124fad3e3e90325a6d1a77829e8a148178b82) this gets restored.

## Related issues

closes #10602 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: .ci/scripts/release/maven-release.sh [hunk 1]
```diff
@@ -11,5 +11,6 @@ mvn -s ${MAVEN_SETTINGS_XML} release:prepare release:perform -B \
     -DpushChanges=${PUSH_CHANGES} \
     -DremoteTagging=${PUSH_CHANGES} \
     -DlocalCheckout=${SKIP_DEPLOY} \
+    -DcompletionGoals="spotless:apply" \
     -P!autoFormat \
-    -Darguments='--settings=${MAVEN_SETTINGS_XML} -P-autoFormat -DskipChecks=true -DskipTests=true -Dgpg.passphrase="${GPG_PASS}" -Dskip.central.release=${SKIP_DEPLOY} -Dskip.camunda.release=${SKIP_DEPLOY} -Dzbctl.force -Dzbctl.rootDir=${ZBCTL_ROOT_DIR}'
+    -Darguments='--settings=${MAVEN_SETTINGS_XML} -P-autoFormat -DskipTests=true -Dgpg.passphrase="${GPG_PASS}" -Dskip.central.release=${SKIP_DEPLOY} -Dskip.camunda.release=${SKIP_DEPLOY} -Dzbctl.force -Dzbctl.rootDir=${ZBCTL_ROOT_DIR}'
```
Comment (korthout) [inline .ci/scripts/release/maven-release.sh:16]: ❌ Why did you re-enable the checks? That would slow down the release process. We also skip the tests. The assumption is that the release is checked and tested in the steps already before the maven release is published.

**Review by korthout — CHANGES_REQUESTED** [review]
Nice work @remcowesterhoud 

👍 I like the addition of the completion goal

❌ There's one blocker, please have a look

File: .ci/scripts/release/maven-release.sh (hunk 1 above)
Comment (remcowesterhoud) [reply .ci/scripts/release/maven-release.sh:16]: > The comments https://github.com/camunda/zeebe/issues/10602#issuecomment-1329376912 mention the deletion of the -DskipChecks=true

Tl;dr, `skipChecks` means spotless gets skipped.

Comment (remcowesterhoud) [reply .ci/scripts/release/maven-release.sh:16]: 🤔 Let me try something else. Maybe I can change the goal to `spotless:apply -DskipChecks=false' to still skip it during the process, but not skipping spotless.

Comment (remcowesterhoud) [reply .ci/scripts/release/maven-release.sh:16]: Unfortunately that doesn't work, so I think it's worth removing the `skipChecks`. The time saved is pretty minor either way.

Comment (korthout) [reply .ci/scripts/release/maven-release.sh:16]: Aha, that makes sense. I missed this line from the PR description.

🔧 It would be helpful to split it off into a separate commit explaining why it's necessary.

🔧 to keep the speed the same, we could also explicitly skip each of the unwanted checks:
```
-Dcheckstyle.skip=true
-Drevapi.skip=true
-Dlicense.skip=true
-Denforcer.skip=true
-Dmdep.analyze.skip=true
```

🔧 Another idea is to disable the skipping of the wanted plugins explicitly:
```
-DskipChecks=true
-Dsort.skip=false
-Dspotless.apply.skip=false
-Dspotless.checks.skip=false
```

Comment (remcowesterhoud) [reply .ci/scripts/release/maven-release.sh:16]: > to keep the speed the same, we could also explicitly skip each of the unwanted checks instead of not skipping any checks:

We use the `-DskipChecks` in so many places. I really don't want to modify them all to skip all plugins independently. It's also annoying if we add a new check because we'd need to update commands everywhere

> Another idea is to disable the skipping of the wanted plugins explicitly, but I'm not sure whether this will work because these values are defined in the parent/pom.xml through skipChecks:

I'll give it a try with adding '-Dspotless.apply.skip=false'. I could see why that would work.

Comment (korthout) [reply .ci/scripts/release/maven-release.sh:16]: 💭 I just noticed that the `sort.skip` is no longer used 💡

Comment (remcowesterhoud) [reply .ci/scripts/release/maven-release.sh]: `-Dspotless.apply.skip=false` seems to work.

The build step is ~2min faster than without: 
image

The formatting still gets applied before the commit:
https://github.com/camunda/zeebe/commit/109d18f4ea4d6f4ac655067caaba7fce05458110

Comment (remcowesterhoud) [reply .ci/scripts/release/maven-release.sh]: I've also added some explanation surrounding it to the commit message.

**Review by korthout — APPROVED** [review]
Thanks for making the changes @remcowesterhoud 

🔧 Please adjust the commit message to our guidelines

**Comment by remcowesterhoud** [comment]
Mm how did that happen. Thanks for notifying me about that.

bors merge
